src/NJsonSchema.Tests/References/LocalReferencesTests.cs

```diff
@@ -123,9 +123,51 @@ public async Task When_reference_is_registered_in_custom_resolver_it_should_not_
             Assert.NotNull(schema);
         }
 
+        [Theory]
+        [InlineData("b%23r", Skip = "Not working ATM")] // Escaped well-formed JSON Pointer
```

Comment:
I did find this is currently not working, I guess I can fix as well but I this is kind-of different issue ;-)

Reply:
Should this even be supported?

Reply:
Let's fix this in another PR.

Reply:
Should I try to fix it?

Reply:
Should this be supported?

Reply:
I guess so...

This is the correct way special characters should be escaped in JSON Pointer and the same applies for JSON Schema/Swagger no?

As https://tools.ietf.org/html/rfc6901#section-6 states:

> while percent-encoding those characters not allowed by the fragment

I did test other validators and they support both percent encoding and special characters (like `#`) in the JSON Pointer part of URI (even though this is not exactly according to spec). In other words if you want to be as compatible as possible, supporting this is a way to go.

Reply:
Ok then we probably should support that and do the split on # and the other sequence everywhere. Do you want to create a PR?

Reply:
We're still talking about `src/NJsonSchema/JsonReferenceResolver.cs` right? If so then sure, I can make the adjustments...

Reply:
Yep exactly

Reply:
Ok, will do